crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -223,6 +224,25 @@ impl SimpleFaucet {
         }))
     }
 
+    /// Clear the WAL list in the faucet
+    pub async fn clear_wal_coins(&self) -> Result<(), FaucetError> {
+        let log = self.wal.lock().await.log.clone();
+        let mut pending = vec![];
+
+        for item in log.safe_iter() {
```

Comment:
`safe_iter` takes the `DBMap` by reference, so I don't think you need to clone it (although maybe I'm missing something).  Generally the `clone` feels strange to me, because `DBMap` is the interface to a Rocksdb database, so I don't know what it means to clone it.

Reply:
move occurs because value has type `DBMap<ObjectID, write_ahead_log::Entry>`, which does not implement the `Copy` traitrustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message%20%5B0%5D?0#file%3A%2F%2F%2FUsers%2Fjianlu%2Fworkspace%2Fsui%2Fcrates%2Fsui-faucet%2Fsrc%2Ffaucet%2Fsimple_faucet.rs)

Reply:
You'll need to share more of the error.  I.e. what is causing the move?

Reply:
error[E0507]: cannot move out of dereference of `tokio::sync::MutexGuard<'_, write_ahead_log::WriteAheadLog>`

Reply:
Got it, (although in future, probably best to paste the whole error in one go).  I think you can get around this by changing:

```rust
let log = /* ... */.log.clone();
```

To this:

```rust
let log = &/* ... */.log;
```